Modules/PostFilter/ColorBreakdown/ColorBreakdownMixingFilterBase.cs

```diff
@@ -39,6 +39,7 @@ internal abstract class ColorBreakdownMixingFilterBase : IntentStateDispatch, IB
 		/// <returns>Intensity for the color component</returns>
 		public double GetIntensityForState(IIntentState intentValue)
 		{
+			_intensityValue = 0;
```

Comment:
The two handlers in this class both assign a value to _intensityValue, so I don't see where this has any effect. I looked at the implementing classes and I did not see any additional handlers that may not be assigning a value. In reading the bug report it seems like this may originate in the fixture feature. Are there additional handlers there that are not assigning a value? The top level processing would set a zero value if there are no intents to be processed so the only path I see for a stale value would be for intents coming through that do not have a handler implementation. Is that something that should be expected? If so are those expected to be swallowed by this filter since they don't have a handler?

I am sure I am missing something here.

Reply:
I think the fixture work exposed this bug but I think it has always been there.   Probably not an issue for pixels or light nodes because what intents other than pixels are going to be present?

Some of the other filters seem to have logic in them to determine if they support an intent.  This filter doesn't seem to have that logic.   The base class has handle methods for all the other intents but they don't clear _intensityValue.

The method ProcessInputData is going to call filter.GetIntensityForState regardless if the handle method is empty or contains logic to set _intensityValue.

I targeted this ticket at the main branch because this issue is present in the main branch but it is unlikely to be stimulated for exposed.

Hopefully some of this will make sense.

Reply:
I don't think it is going to clear the state.  ProcessInputData is looping over all the intents and taking the maximum intensity.

The fixture has one node getting all the intents for the fixture (range, index, color intents).  The filter is holding onto last frame's color data.  It gets one of these intents it doesn't intend to handle and then returns the last color value it had.

public double GetIntensityForState(IIntentState intentValue)
{
    intentValue.Dispatch(this);
     return _intensityValue;     // <-- What code checked to make sure this value is legit for this frame?
}